### DISCUSSION TITLE
using OpenRadioss without starter, or building starter without hypermesh reader binary blob?

### DISCUSSION THREAD

**Discussion by ky438** [OpenRadioss/OpenRadioss#966] [Ideas]
Hello, @OlivierWienholtz 

I managed to get OpenRadioss **engine** working on linux ppc64le (IBM POWER) system, it was simple.

However, OpenRadioss **starter** does not build, because it seems to depend on a binary blob (libhm_reader) that is not available for this platform.

Is it possible to use OpenRadioss without the starter, or is it possible to build a 'simplified' starter that does not depend on libhm_reader? (Or, is it possible to get a version of libhm_reader for linux ppc64le?)

Is independently developing an open source 'starter' a reasonable project for one person or the community to undertake? Or, is there any plan to open source the code behind libhm_reader?

**Comment by elequiniou** [comment]
Dear ky438,
First, really nice to learn you succeeded to port on the IBM Power (Radioss used to compile & run on IBM Power but time ago...)! As a first workaround, I'd suggest you run the starter on a standard Linux or Windows system and then use the restart(s) on the IBM PPC machine to run the engine. Then, at least you will be able to use your PPC machine for the most time consuming and scalable part of the simulation... The time we find a solution to port the libhm_reader on such platform. Thanks!

**Comment by ky438** [comment]
Dear Eric,

Yes, the PPC machine is fast and so this workaround is useful. At some point, I will try to contribute the changes required to build the engine.

It would be nice to have libhm_reader open sourced, or maybe the restart file format documented so that it is possible for users to interact with the engine directly, or to write their own simple "starter". If you need access to a Linux ppc64le machine for porting libhm_reader, please let me know as I should be able to assist.

Regards,
Kumi

**Comment by elequiniou** [comment]
Dear Kumi,

Thanks for your follow-up and effort regarding PPC porting!
Unfortunately, this is not possible to open source the libhm_reader as it is today. This is a point that can be reconsidered in the future, as we are actively working on providing an improved API to this library.
As it is not something to be changed right away, I propose to convert this issue into a discussion, if you don't mind.

Thanks & Best Regards,

Eric.

**Comment by ky438** [comment]
Dear Eric,

It is wonderful to learn that you are actively working on an improved API to libhm_reader and that perhaps one day it might be possible to open source.

Of course I do not mind converting this issue into a discussion, please go ahead.

Regards,
Kumi

**Comment by ky438** [comment]
Some years have passed - is there any update on open-sourcing / reworking libhm_reader @elequiniou ?

**Reply by SyazwanSamad** [reply]
If not mistaken, we can now use open_reader as alternative, instead of hm_reader (compile the starter with -open_reader argument)

**Comment by sebastienVilleneuve** [comment]
Hi @ky438 

https://github.com/OpenRadioss/OpenRadioss/blob/main/HOWTO.md
<img width="1621" height="949" alt="image" src="https://github.com/user-attachments/assets/76646d1d-3788-4cd6-ac36-9ee5100ef7f6" />

This is not plugged as the default reader for the moment.
Don't hesitate to report issue , if you encounter problems .

Regards,
Sébastien
